app/models/service_retire_task.rb

```diff
@@ -42,22 +42,22 @@ def create_retire_subtasks(parent_service, parent_task)
       # Initial Options[:dialog] to an empty hash so we do not pass down dialog values to child services tasks
       nh['options'][:dialog] = {}
       new_task = create_task(svc_rsc, parent_service, nh, parent_task)
-      create_retire_subtasks(svc_rsc.resource, new_task) if svc_rsc.resource.kind_of?(Service)
       new_task.after_request_task_create
       miq_request.miq_request_tasks << new_task
       new_task.tap(&:deliver_to_automate)
     end.compact!
   end
 
   def create_task(svc_rsc, parent_service, nh, parent_task)
-    (svc_rsc.resource.type.demodulize + "RetireTask").constantize.new(nh).tap do |task|
+    resource_type = svc_rsc.resource.type.presence || "Service"
```

Comment:
`type` can't be blank, can it?

Reply:
The subservice of a bundle will have a nil type.